warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyExcerptIterator.java

```diff
@@ -278,6 +300,16 @@ public void next() throws IOException {
                 try {
                     // parse the offsets from the value
                     TermWeight.Info info = TermWeight.Info.parseFrom(source.getTopValue().get());
+                    // check if the number of scores is equal to the number of offsets
+                    boolean useScores = info.getScoreCount() == info.getTermOffsetCount();
+                    boolean hasOnlyNegativeScores = true;
+                    List<Integer> scoreList = info.getScoreList();
+                    for (Integer integer : scoreList) {
```

Comment:
it appears that you only need to check the scores if `useScores` is true. Consider setting `useScores` to false here if you find out that all scores are negative. This simplifies the check below. 

Also, don't we only want to look at the scores for the terms between our start and end offset? I suppose there could be positive scores only outside of the range of offsets we're interested in, and if we only consider the scores for the terms between our start and end offset there's a possibility that they will all be negative.

Reply:
both done